File: java/src/main/java/com/meteroid/models/SubscriptionUpdateType.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum SubscriptionUpdateType implements ToQueryParam {
    ACTIVATED("activated"),
    TRIAL_ENDED("trial_ended"),
    BILLING_CONFIGURATION_UPDATED("billing_configuration_updated"),
    PLAN_CHANGED("plan_changed"),
    AMENDED("amended"),
    UNITS_CHANGED("units_changed"),
    PAUSED("paused"),
    CANCELLATION_SCHEDULED("cancellation_scheduled");
    private final String value;

    SubscriptionUpdateType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: python/meteroid/models/subscription_update_type.py
# this file is @generated
import enum


class SubscriptionUpdateType(str, enum.Enum):
    """Identifies which mutation triggered a `subscription.updated` webhook."""

    ACTIVATED = "activated"
    TRIAL_ENDED = "trial_ended"
    BILLING_CONFIGURATION_UPDATED = "billing_configuration_updated"
    PLAN_CHANGED = "plan_changed"
    AMENDED = "amended"
    UNITS_CHANGED = "units_changed"
    PAUSED = "paused"
    CANCELLATION_SCHEDULED = "cancellation_scheduled"

    def __str__(self) -> str:
        return str(self.value)
